From 803b2f58760ed82e5eb182df75ba18c00c9f43bf Mon Sep 17 00:00:00 2001
From: Stefano Lattarini <stefano.lattarini@gmail.com>
Date: Thu, 30 Aug 2012 18:53:16 +0200
Subject: [PATCH 13/89] build: enable subdir-objects Automake option globally

That option is enabled unconditionally in Automake-NG, and enabling
it also for mainline Automake will help to ensure no unexpected
incompatibilities is introduced in a potential port to Automake-NG.

Moreover, if we have a source file 'sub/foo.c', having it compiled
in 'sub/foo.o' rather than in 'foo.o' is undeniably cleaner and more
natural.

* configure.ac (AM_INIT_AUTOMAKE): Add 'subdir-objects'.
* libparted/fs/Makefile.am (AUTOMAKE_OPTIONS): No need to declare
'subdir-objects' explicitly now.

Copyright-paperwork-exempt: yes
---
 configure.ac             | 3 ++-
 libparted/fs/Makefile.am | 2 --
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 1444279..49ef75c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -52,7 +52,8 @@ AC_SUBST([LT_CURRENT])
 AC_SUBST([LT_REVISION])
 AC_SUBST([LT_AGE])
 
-AM_INIT_AUTOMAKE([1.11 no-dist-gzip dist-xz color-tests parallel-tests])
+AM_INIT_AUTOMAKE([1.11 no-dist-gzip dist-xz color-tests parallel-tests
+                  subdir-objects])
 AM_SILENT_RULES([yes]) # make --enable-silent-rules the default.
 
 AC_CANONICAL_HOST
diff --git a/libparted/fs/Makefile.am b/libparted/fs/Makefile.am
index 7fe364a..064f11a 100644
--- a/libparted/fs/Makefile.am
+++ b/libparted/fs/Makefile.am
@@ -3,8 +3,6 @@
 #
 # This file may be modified and/or distributed without restriction.
 
-AUTOMAKE_OPTIONS = subdir-objects
-
 partedincludedir = -I$(top_builddir)/include -I$(top_srcdir)/include
 
 AM_CFLAGS = $(WARN_CFLAGS)
-- 
1.8.5.3

